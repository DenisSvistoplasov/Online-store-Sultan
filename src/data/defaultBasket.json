{"productsIds":{"5":1,"9":2,"12":1},"totalCount":4}