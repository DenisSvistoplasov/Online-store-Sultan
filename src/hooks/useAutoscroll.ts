import { useEffect } from "react";

export const useAutoscroll = (container: HTMLElement | null, refs: any[]) => {
  useEffect(() => {
    if (container && container.getBoundingClientRect().bottom < 0) {
      container.scrollIntoView({ behavior: 'smooth', block: 'end' });
    }
  }, refs);
};