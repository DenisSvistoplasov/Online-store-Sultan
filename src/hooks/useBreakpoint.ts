import { useContext } from "react";
import { breakpointContext } from "../context/BreakpointContext";

export const useBreakpoint = () => {
  return useContext(breakpointContext);
};