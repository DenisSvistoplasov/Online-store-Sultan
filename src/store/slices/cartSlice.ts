import { PayloadAction, createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { IProduct } from "../../data/catalog";
import { RootState } from "../store";
import defaultBasket from "../../data/defaultBasket.json";

interface ICartState {
  productsIds: { [id: string]: number; };
  totalCount: number;
}

const LOCAL_STORAGE_KEY = 'cart-data';

const localCart = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY) || 'null');

const initialState = localCart as ICartState || defaultBasket;

const saveLocally = (state: ICartState) => localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(state));

const cartSlice = createSlice({
  name: 'cart',
  initialState,
  reducers: {
    plusToCart(state, action: PayloadAction<string | undefined>) {
      if (!action.payload) return;
      if (!state.productsIds[action.payload]) state.productsIds[action.payload] = 0;
      state.productsIds[action.payload]++;
      state.totalCount++;
      saveLocally(state);
    },
    minusFromCart(state, action: PayloadAction<string | undefined>) {
      if (!action.payload) return;
      if (state.productsIds[action.payload]) {
        state.productsIds[action.payload]--;
        state.totalCount--;
        if (state.productsIds[action.payload] === 0) delete state.productsIds[action.payload];
        saveLocally(state);
      }
    },
    removeFromCart(state, action: PayloadAction<string | undefined>) {
      if (!action.payload) return;
      if (state.productsIds[action.payload]) {
        state.totalCount -= state.productsIds[action.payload];
        state.productsIds[action.payload] = 0;
        delete state.productsIds[action.payload];
        saveLocally(state);
      }
    },
    clearCart(state) {
      state.productsIds = {};
      state.totalCount = 0;
      saveLocally(state);
    },
  }
});

export default cartSlice.reducer;

export const { plusToCart, minusFromCart, removeFromCart, clearCart } = cartSlice.actions;

export const selectCartProductsIds = (state: RootState) => state.cart.productsIds;
export const selectCartTotalCount = (state: RootState) => state.cart.totalCount;
export const selectProductInCartCount = (id?: string) => (state: RootState) => id ? (state.cart.productsIds[id] || 0) : 0;
export const selectCartTotalCost = (state: RootState) => {
  const catalog = state.products.data;
  const cartProductsIds = state.cart.productsIds;
  let totalCost = 0;

  for (const key in cartProductsIds) {
    let cost = catalog.find(product => product.id == key)?.price || 0;
    totalCost += cost * cartProductsIds[key];
  }

  return totalCost;
};
export const selectCartRepresentation = (state: RootState) => {
  const isLoading = state.products.meta.status === 'loading';
  const catalog = state.products.data;
  const cartProductsIds = state.cart.productsIds;

  const cartProducts = [] as (IProduct & { count: number; cost: number; })[];
  let totalCost = 0;

  if (!isLoading) {
    for (const id in cartProductsIds) {
      const count = cartProductsIds[id];
      const product = catalog.find(product => product.id == id);

      if (!product) throw new Error('Product in basket does not exist in catalog (id: ' + id + ')');

      const cost = product.price * count;
      cartProducts.push({ ...product, count, cost });
      totalCost += cost;
    }
  }

  return {
    products: cartProducts,
    totalCost
  };
};