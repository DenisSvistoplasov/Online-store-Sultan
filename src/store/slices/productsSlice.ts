import { PayloadAction, SerializedError, createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { IProduct } from "../../data/catalog";
import { fetchStatus } from "../../types/types";
import { RootState } from "../store";

const LOCAL_STORAGE_KEY = 'products-data';

export const setProducts = createAsyncThunk<IProduct[]>(
  'products/getProducts',
  async () => {
    let data = localStorage.getItem(LOCAL_STORAGE_KEY);

    if (data) return JSON.parse(data);
    return (await import('../../data/catalog')).default;
  });



const productsSlice = createSlice({
  name: 'products',
  initialState: {
    data: [] as IProduct[],
    meta: {
      status: '' as fetchStatus,
      error: '' as string | SerializedError
    }
  },
  reducers: {},
  extraReducers(builder) {
    builder
      .addCase(setProducts.pending, (state, action) => {
        state.meta.status = "loading";
      })
      .addCase(setProducts.fulfilled, (state, action) => {
        state.meta.status = "succeeded";
        state.data = action.payload;
      })
      .addCase(setProducts.rejected, (state, action) => {
        state.meta.status = "failed";
        state.meta.error = action.error;
      });
  }
});
export default productsSlice.reducer;

export const selectProducts = (state: RootState) => state.products.data;
export const selectBrandsAndManufacturers = (state: RootState) => {
  const res = {
    brands: [] as { value: string; count: number; }[],
    manufacturers: [] as { value: string; count: number; }[]
  };
  const brands = {} as { [key: string]: number; };
  const manufacturers = {} as { [key: string]: number; };

  state.products.data.forEach(product => {
    if (!brands[product.brand]) brands[product.brand] = 0;
    if (!manufacturers[product.manufacturer]) manufacturers[product.manufacturer] = 0;
    brands[product.brand]++;
    manufacturers[product.manufacturer]++;
  });

  for (const key in brands) {
    res.brands.push({ value: key, count: brands[key] });
  }
  for (const key in manufacturers) {
    res.manufacturers.push({ value: key, count: manufacturers[key] });
  }

  return res;
};
