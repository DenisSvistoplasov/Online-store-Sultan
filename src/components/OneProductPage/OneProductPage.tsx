import styles from './oneproductpage.sass';
import { useParams } from 'react-router-dom';
import { Container } from '../Container';
import { IconBox } from '../icons/box';
import { IconBottle } from '../icons/bottle';
import { IconCart } from '../icons/cart';
import { IconShare } from '../icons/share';
import { IconDownload } from '../icons/download';
import { useAppDispatch, useAppSelector } from '../../hooks/storeHooks';
import { plusToCart, minusFromCart, selectProductInCartCount, removeFromCart } from '../../store/slices/cartSlice';
import { classnames } from '../../utils/classnames';
import { selectProductsByIds, selectProductsStatus } from '../../store/slices/productsSlice';
import { Counter } from '../Counter';
import { fixNumber } from '../../utils/fixNumber';
import { InfoMessage } from '../InfoMessage';
import { Dropdown } from '../Dropdown';



export function OneProductPage() {
  const isLoading = useAppSelector(selectProductsStatus) === 'loading';
  const id = useParams().id;
  const productData = useAppSelector(selectProductsByIds([id]))[0];
  const count = useAppSelector(selectProductInCartCount(id));
  const dispatch = useAppDispatch();

  if (isLoading) return <>Loading...</>;
  if (!productData) return <InfoMessage text='Товар не найден' />;

  const {
    imageUrl,
    name,
    quantityType,
    amount,
    barcode,
    manufacturer,
    brand,
    description,
    price,
    fullDescription,
    types: type,
    isPopular,
  } = productData;

  const onPlus = () => dispatch(plusToCart(id));
  const onMinus = () => dispatch(minusFromCart(id));
  const onRemove = () => dispatch(removeFromCart(id));

  return (
    <section className={styles.section}>
      <Container className={styles.container}>

        <div className={styles["img-wrapper"]}>
          {isPopular && <div className={styles.popular}>ПОПУЛЯРНОЕ</div>}
          <img src={imageUrl} alt={name} className={styles.img} />
        </div>

        <div className={styles["text-content"]}>

          <div className={styles["in-stock"]}>В наличии</div>
          <h3 className={styles.description}><span className={styles.name}>{name}</span> {description}</h3>
          <div className={styles.amount}>
            {quantityType == 'вес' && <IconBox />}
            {quantityType == 'объем' && <IconBottle />}
            {amount}
          </div>

          <div className={styles.controls}>
            <div className={styles.price}>{fixNumber(price)} ₸</div>
            <Counter className={styles.counter} count={count} onMinus={onMinus} onPlus={onPlus} />
            {count ?
              <button className={classnames(styles.remove, 'interactive-btn')} onClick={onRemove}>Убрать<IconCart /></button> :
              <button className={classnames(styles.add, 'interactive-btn')} onClick={onPlus}>В корзину<IconCart /></button>
            }
            <div className={styles.share}><IconShare /></div>
            <div className={styles['promotion-and-price-list']}>
              <p className={styles.promotion}>При покупке от <b>10 000 ₸</b> бесплатная доставка по Кокчетаву и области</p>
              <a href="" className={styles["price-list-link"]}>Прайс-лист<IconDownload /></a>
            </div>
          </div>

          <div className={styles.manufacturer}>Производитель:<b>{manufacturer}</b></div>
          <div className={styles.brand}>Бренд:<b>{brand}</b></div>
          <div className={styles.barcode}>Штрихкод:<b>{barcode}</b></div>

          <Dropdown button='Описание'>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam interdum ut justo, vestibulum sagittis iaculis iaculis. Quis mattis vulputate feugiat massa vestibulum duis. Faucibus consectetur aliquet sed pellentesque consequat consectetur congue mauris venenatis. Nunc elit, dignissim sed nulla ullamcorper enim, malesuada.</Dropdown>
          <div className={styles.delimiter} />
          <Dropdown button='Характеристики'>
            <div className={styles.manufacturer}>Производитель:<b>{manufacturer}</b></div>
            <div className={styles.brand}>Бренд:<b>{brand}</b></div>
            <div className={styles.barcode}>Штрихкод:<b>{barcode}</b></div>
          </Dropdown>
        </div>

      </Container>
    </section>
  );
}
