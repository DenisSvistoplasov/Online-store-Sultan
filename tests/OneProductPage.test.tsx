import './matchMedia';
import * as storeHooks from "../src/hooks/storeHooks";
import { RenderResult, fireEvent, render, screen } from '@testing-library/react';
import { OneProductPage } from "../src/components/OneProductPage";
import { AllProviders } from "./testUtils";
import * as productSlice from '../src/store/slices/productsSlice';
import { IProduct } from '../src/data/catalog';
import * as routerDom from 'react-router-dom';
import * as cartSlice from '../src/store/slices/cartSlice';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ id: '99' }),
}));

const selectProductsStatus = jest.spyOn(productSlice, 'selectProductsStatus');
const selectProductsByIds = jest.spyOn(productSlice, 'selectProductsByIds');
const useParams = routerDom.useParams;
const selectProductInCartCount = jest.spyOn(cartSlice, 'selectProductInCartCount');
const useDispatch = jest.spyOn(storeHooks, 'useAppDispatch');
const dispatch = jest.fn();

const product = {
  imageUrl: 'aosImg',
  name: 'AOS',
  quantityType: 'объем',
  amount: '450 мл',
  description: 'средство для мытья посуды Crystal',
  fullDescription: 'Далеко-далеко за словесными горами в стране гласных и согласных живут рыбные тексты. Использовало составитель, до ее безопасную рукописи за риторический скатился ему имеет предупреждал собрал по всей заголовок решила жизни, запятой его буквенных агентство если свое над переписали. Собрал гор журчит домах необходимыми.',
  barcode: '4604049097548',
  manufacturer: 'Нэфис',
  brand: 'AOS',
  price: 48.76,
  id: '0',
  types: ['body'],
  isPopular: true
} as IProduct;

selectProductsStatus.mockReturnValue('succeeded');
selectProductsByIds.mockReturnValue(() => [product]);
selectProductInCartCount.mockReturnValue(() => 0);
useDispatch.mockReturnValue(dispatch);
let component: RenderResult;

beforeEach(() => {
  selectProductsStatus.mockReturnValue('succeeded');
  selectProductsByIds.mockReturnValue(() => [product]);
  selectProductInCartCount.mockReturnValue(() => 0);
  useDispatch.mockReturnValue(dispatch);
  component = render(
    <AllProviders>
      <OneProductPage />
    </AllProviders>
  );
  
});
describe('OneProductPage', () => {
  it('Render', () => {
    expect(component).toMatchSnapshot();
  });
  it('Check product status', () => {
    expect(selectProductsStatus).toBeCalled();
  });
  it('Get id from URL', () => {
    expect(useParams).toBeCalled();
  });
  it('Get product data by the id', () => {
    expect(selectProductsByIds).toBeCalledWith(['99']);
  });
  it('Get products count by the id', () => {
    expect(selectProductInCartCount).toBeCalledWith('99');
  });
  it('Dispatch plusToCart on button click', async () => {
    let addBtn = await screen.findByTestId('addBtn');
    fireEvent.click(addBtn as Element);
    expect(dispatch).toBeCalledWith(cartSlice.plusToCart('99'));
  });
});