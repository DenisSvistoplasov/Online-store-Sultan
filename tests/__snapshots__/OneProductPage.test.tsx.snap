// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneProductPage Render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class=""
        >
          <div>
            <div>
              ПОПУЛЯРНОЕ
            </div>
            <img
              alt="AOS"
              src="aosImg"
            />
          </div>
          <div>
            <div>
              В наличии
            </div>
            <h3>
              <span>
                AOS
              </span>
               
              средство для мытья посуды Crystal
            </h3>
            <div>
              <svg
                fill="none"
                height="15"
                opacity="0.2"
                viewBox="0 0 9 15"
                width="9"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.1 14.0C8.0 14.3 7.9 14.5 7.7 14.7C7.6 14.9 7.4 15 7.2 15H1.8C1.5 15 1.3 14.9 1.2 14.7C1.0 14.5 0.9 14.3 0.9 14.0L0 8.4V6.5C0 6.2 0.0 6.0 0.2 5.8C0.4 5.6 0.7 5.4 1.0 5.3C1.4 5.1 1.6 5.0 1.7 4.9C2.1 4.6 2.4 4.3 2.7 3.9C3.0 3.6 3.2 3.2 3.4 2.8H3.1C3.0 2.8 2.9 2.7 2.8 2.6C2.7 2.5 2.7 2.4 2.7 2.3V0.4C2.7 0.3 2.7 0.2 2.8 0.1C2.9 0.0 3.0 0 3.1 0H5.8C5.9 0 6.0 0.0 6.1 0.1C6.2 0.2 6.3 0.3 6.3 0.4V2.3C6.3 2.4 6.2 2.5 6.1 2.6C6.0 2.7 5.9 2.8 5.8 2.8H5.5C5.8 3.6 6.3 4.2 7.1 4.8C7.2 4.9 7.4 5.0 7.8 5.2C8.2 5.4 8.4 5.6 8.6 5.8C8.8 6.0 9 6.2 9 6.5V8.4L8.1 14.0Z"
                  fill="#3F4E65"
                />
              </svg>
              450 мл
            </div>
            <div>
              <div>
                48,76
                 ₸
              </div>
              <div
                class=""
              >
                <button>
                  -
                </button>
                <div>
                  0
                </div>
                <button>
                  +
                </button>
              </div>
              <button
                class="interactive-btn"
                data-testid="addBtn"
              >
                В корзину
                <svg
                  fill="none"
                  height="25"
                  viewBox="0 0 25 25"
                  width="25"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22.6 7.4C22.4 7.2 22.3 7.1 22.0 7.1H6.9L6.5 6.1C6.4 5.7 6.2 5.4 6.0 5.2C5.9 4.9 5.7 4.8 5.5 4.7C5.3 4.6 5.1 4.5 5.0 4.4C4.8 4.4 4.7 4.4 4.5 4.4H1.5C1.3 4.4 1.1 4.5 1.0 4.6C0.8 4.7 0.8 4.9 0.8 5.1C0.8 5.2 0.8 5.3 0.9 5.4C0.9 5.5 1.0 5.6 1.1 5.7C1.2 5.7 1.3 5.8 1.5 5.8H4.5C4.6 5.8 4.6 5.8 4.7 5.8C4.7 5.8 4.8 5.9 4.9 6.0C5.0 6.1 5.1 6.3 5.1 6.5L8.3 15.3C8.3 15.3 8.4 15.4 8.5 15.5C8.5 15.6 8.6 15.6 8.7 15.7C8.8 15.7 8.9 15.8 9.0 15.8H18.6C18.8 15.8 18.9 15.7 19.0 15.6C19.1 15.5 19.2 15.4 19.3 15.3L22.7 8.1C22.8 7.9 22.7 7.6 22.6 7.4ZM18.1 14.4H9.5L7.3 8.5H21.0L18.1 14.4ZM16.8 16.5C16.3 16.5 15.9 16.7 15.6 17.0C15.2 17.4 15.0 17.8 15.0 18.3C15.0 18.8 15.2 19.2 15.6 19.5C15.9 19.8 16.3 20.0 16.8 20.0C17.3 20.0 17.7 19.8 18.1 19.5C18.4 19.2 18.6 18.8 18.6 18.3C18.6 17.8 18.4 17.4 18.1 17.0C17.7 16.7 17.3 16.5 16.8 16.5ZM10.4 16.5C10.1 16.5 9.8 16.6 9.5 16.8C9.2 16.9 9.0 17.1 8.9 17.4C8.7 17.7 8.6 17.9 8.6 18.3C8.6 18.8 8.8 19.2 9.2 19.5C9.5 19.8 9.9 20.0 10.4 20.0C10.9 20.0 11.3 19.8 11.7 19.5C12.0 19.2 12.2 18.8 12.2 18.3C12.2 18.2 12.2 18.0 12.2 17.9C12.1 17.8 12.1 17.7 12.1 17.6C12.0 17.5 12.0 17.4 11.9 17.3C11.8 17.2 11.8 17.1 11.7 17.0C11.6 16.9 11.5 16.9 11.4 16.8C11.3 16.7 11.2 16.7 11.1 16.6C11.0 16.6 10.9 16.6 10.8 16.5C10.6 16.5 10.5 16.5 10.4 16.5Z"
                    fill="#3F4E65"
                  />
                </svg>
              </button>
              <div>
                <svg
                  fill="none"
                  height="25"
                  viewBox="0 0 25 25"
                  width="25"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.0 15.5C6.8 15.4 7.7 15.1 8.3 14.5L14.6 18.1C14.4 18.9 14.4 19.8 14.8 20.5C15.2 21.3 15.8 21.9 16.6 22.2C17.4 22.5 18.3 22.6 19.1 22.3C19.9 22.0 20.6 21.5 21.0 20.8C21.4 20.0 21.6 19.2 21.4 18.4C21.3 17.5 20.9 16.8 20.2 16.2C19.5 15.7 18.7 15.4 17.9 15.5C17.0 15.5 16.2 15.8 15.6 16.4L9.3 12.8C9.4 12.6 9.4 12.3 9.4 12.0L15.6 8.5C16.2 9.1 16.9 9.4 17.7 9.4C18.5 9.5 19.3 9.2 20.0 8.8C20.6 8.3 21.1 7.6 21.3 6.8C21.5 6.1 21.5 5.3 21.1 4.5C20.8 3.8 20.2 3.2 19.5 2.8C18.8 2.5 18.0 2.4 17.2 2.5C16.4 2.7 15.7 3.1 15.2 3.7C14.7 4.4 14.4 5.1 14.5 5.9C14.5 6.2 14.5 6.5 14.6 6.8L8.9 10.1C8.6 9.5 8.1 9.1 7.6 8.8C7.0 8.6 6.4 8.4 5.8 8.5C5.2 8.5 4.6 8.7 4.1 9.0C3.6 9.3 3.2 9.7 2.9 10.3C2.6 10.8 2.4 11.4 2.4 12.0C2.5 12.6 2.6 13.2 2.9 13.7C3.2 14.3 3.7 14.7 4.2 15.0C4.7 15.3 5.3 15.5 6.0 15.5V15.5Z"
                    fill="#FFC85E"
                  />
                </svg>
              </div>
              <div>
                <p>
                  При покупке от 
                  <b>
                    10 000 ₸
                  </b>
                   бесплатная доставка по Кокчетаву и области
                </p>
                <a
                  href=""
                >
                  Прайс-лист
                  <svg
                    fill="none"
                    height="18"
                    viewBox="0 0 18 18"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.9 6.8H11.1V2.6H6.8V6.8H4.0L9.0 12.5L13.9 6.8ZM3.3 13.9H14.6V15.3H3.3V13.9Z"
                      fill="white"
                    />
                  </svg>
                </a>
              </div>
            </div>
            <div>
              Производитель:
              <b>
                Нэфис
              </b>
            </div>
            <div>
              Бренд:
              <b>
                AOS
              </b>
            </div>
            <div>
              Штрихкод:
              <b>
                4604049097548
              </b>
            </div>
            <div
              class=""
            >
              <button>
                Описание
                <div
                  class=""
                >
                  ▼
                </div>
              </button>
            </div>
            <div />
            <div
              class=""
            >
              <button>
                Характеристики
                <div
                  class=""
                >
                  ▼
                </div>
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class=""
      >
        <div>
          <div>
            ПОПУЛЯРНОЕ
          </div>
          <img
            alt="AOS"
            src="aosImg"
          />
        </div>
        <div>
          <div>
            В наличии
          </div>
          <h3>
            <span>
              AOS
            </span>
             
            средство для мытья посуды Crystal
          </h3>
          <div>
            <svg
              fill="none"
              height="15"
              opacity="0.2"
              viewBox="0 0 9 15"
              width="9"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.1 14.0C8.0 14.3 7.9 14.5 7.7 14.7C7.6 14.9 7.4 15 7.2 15H1.8C1.5 15 1.3 14.9 1.2 14.7C1.0 14.5 0.9 14.3 0.9 14.0L0 8.4V6.5C0 6.2 0.0 6.0 0.2 5.8C0.4 5.6 0.7 5.4 1.0 5.3C1.4 5.1 1.6 5.0 1.7 4.9C2.1 4.6 2.4 4.3 2.7 3.9C3.0 3.6 3.2 3.2 3.4 2.8H3.1C3.0 2.8 2.9 2.7 2.8 2.6C2.7 2.5 2.7 2.4 2.7 2.3V0.4C2.7 0.3 2.7 0.2 2.8 0.1C2.9 0.0 3.0 0 3.1 0H5.8C5.9 0 6.0 0.0 6.1 0.1C6.2 0.2 6.3 0.3 6.3 0.4V2.3C6.3 2.4 6.2 2.5 6.1 2.6C6.0 2.7 5.9 2.8 5.8 2.8H5.5C5.8 3.6 6.3 4.2 7.1 4.8C7.2 4.9 7.4 5.0 7.8 5.2C8.2 5.4 8.4 5.6 8.6 5.8C8.8 6.0 9 6.2 9 6.5V8.4L8.1 14.0Z"
                fill="#3F4E65"
              />
            </svg>
            450 мл
          </div>
          <div>
            <div>
              48,76
               ₸
            </div>
            <div
              class=""
            >
              <button>
                -
              </button>
              <div>
                0
              </div>
              <button>
                +
              </button>
            </div>
            <button
              class="interactive-btn"
              data-testid="addBtn"
            >
              В корзину
              <svg
                fill="none"
                height="25"
                viewBox="0 0 25 25"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22.6 7.4C22.4 7.2 22.3 7.1 22.0 7.1H6.9L6.5 6.1C6.4 5.7 6.2 5.4 6.0 5.2C5.9 4.9 5.7 4.8 5.5 4.7C5.3 4.6 5.1 4.5 5.0 4.4C4.8 4.4 4.7 4.4 4.5 4.4H1.5C1.3 4.4 1.1 4.5 1.0 4.6C0.8 4.7 0.8 4.9 0.8 5.1C0.8 5.2 0.8 5.3 0.9 5.4C0.9 5.5 1.0 5.6 1.1 5.7C1.2 5.7 1.3 5.8 1.5 5.8H4.5C4.6 5.8 4.6 5.8 4.7 5.8C4.7 5.8 4.8 5.9 4.9 6.0C5.0 6.1 5.1 6.3 5.1 6.5L8.3 15.3C8.3 15.3 8.4 15.4 8.5 15.5C8.5 15.6 8.6 15.6 8.7 15.7C8.8 15.7 8.9 15.8 9.0 15.8H18.6C18.8 15.8 18.9 15.7 19.0 15.6C19.1 15.5 19.2 15.4 19.3 15.3L22.7 8.1C22.8 7.9 22.7 7.6 22.6 7.4ZM18.1 14.4H9.5L7.3 8.5H21.0L18.1 14.4ZM16.8 16.5C16.3 16.5 15.9 16.7 15.6 17.0C15.2 17.4 15.0 17.8 15.0 18.3C15.0 18.8 15.2 19.2 15.6 19.5C15.9 19.8 16.3 20.0 16.8 20.0C17.3 20.0 17.7 19.8 18.1 19.5C18.4 19.2 18.6 18.8 18.6 18.3C18.6 17.8 18.4 17.4 18.1 17.0C17.7 16.7 17.3 16.5 16.8 16.5ZM10.4 16.5C10.1 16.5 9.8 16.6 9.5 16.8C9.2 16.9 9.0 17.1 8.9 17.4C8.7 17.7 8.6 17.9 8.6 18.3C8.6 18.8 8.8 19.2 9.2 19.5C9.5 19.8 9.9 20.0 10.4 20.0C10.9 20.0 11.3 19.8 11.7 19.5C12.0 19.2 12.2 18.8 12.2 18.3C12.2 18.2 12.2 18.0 12.2 17.9C12.1 17.8 12.1 17.7 12.1 17.6C12.0 17.5 12.0 17.4 11.9 17.3C11.8 17.2 11.8 17.1 11.7 17.0C11.6 16.9 11.5 16.9 11.4 16.8C11.3 16.7 11.2 16.7 11.1 16.6C11.0 16.6 10.9 16.6 10.8 16.5C10.6 16.5 10.5 16.5 10.4 16.5Z"
                  fill="#3F4E65"
                />
              </svg>
            </button>
            <div>
              <svg
                fill="none"
                height="25"
                viewBox="0 0 25 25"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.0 15.5C6.8 15.4 7.7 15.1 8.3 14.5L14.6 18.1C14.4 18.9 14.4 19.8 14.8 20.5C15.2 21.3 15.8 21.9 16.6 22.2C17.4 22.5 18.3 22.6 19.1 22.3C19.9 22.0 20.6 21.5 21.0 20.8C21.4 20.0 21.6 19.2 21.4 18.4C21.3 17.5 20.9 16.8 20.2 16.2C19.5 15.7 18.7 15.4 17.9 15.5C17.0 15.5 16.2 15.8 15.6 16.4L9.3 12.8C9.4 12.6 9.4 12.3 9.4 12.0L15.6 8.5C16.2 9.1 16.9 9.4 17.7 9.4C18.5 9.5 19.3 9.2 20.0 8.8C20.6 8.3 21.1 7.6 21.3 6.8C21.5 6.1 21.5 5.3 21.1 4.5C20.8 3.8 20.2 3.2 19.5 2.8C18.8 2.5 18.0 2.4 17.2 2.5C16.4 2.7 15.7 3.1 15.2 3.7C14.7 4.4 14.4 5.1 14.5 5.9C14.5 6.2 14.5 6.5 14.6 6.8L8.9 10.1C8.6 9.5 8.1 9.1 7.6 8.8C7.0 8.6 6.4 8.4 5.8 8.5C5.2 8.5 4.6 8.7 4.1 9.0C3.6 9.3 3.2 9.7 2.9 10.3C2.6 10.8 2.4 11.4 2.4 12.0C2.5 12.6 2.6 13.2 2.9 13.7C3.2 14.3 3.7 14.7 4.2 15.0C4.7 15.3 5.3 15.5 6.0 15.5V15.5Z"
                  fill="#FFC85E"
                />
              </svg>
            </div>
            <div>
              <p>
                При покупке от 
                <b>
                  10 000 ₸
                </b>
                 бесплатная доставка по Кокчетаву и области
              </p>
              <a
                href=""
              >
                Прайс-лист
                <svg
                  fill="none"
                  height="18"
                  viewBox="0 0 18 18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.9 6.8H11.1V2.6H6.8V6.8H4.0L9.0 12.5L13.9 6.8ZM3.3 13.9H14.6V15.3H3.3V13.9Z"
                    fill="white"
                  />
                </svg>
              </a>
            </div>
          </div>
          <div>
            Производитель:
            <b>
              Нэфис
            </b>
          </div>
          <div>
            Бренд:
            <b>
              AOS
            </b>
          </div>
          <div>
            Штрихкод:
            <b>
              4604049097548
            </b>
          </div>
          <div
            class=""
          >
            <button>
              Описание
              <div
                class=""
              >
                ▼
              </div>
            </button>
          </div>
          <div />
          <div
            class=""
          >
            <button>
              Характеристики
              <div
                class=""
              >
                ▼
              </div>
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
